hls4ml/backends/vivado/passes/convolution_templates.py

```diff
@@ -56,6 +56,8 @@
     typedef {config_t} mult_config;
     template<unsigned K, unsigned S, unsigned W>
     using scale_index = nnet::{scale_index_type}<K, S, W>;
+    template<class data_T, class res_T, class CONFIG_T>
```

Comment:
Doesn't this mean that every Conv1D layer will have a pointer to a pointwise function that it may not even use?

Reply:
OK, I moved this only in the pointwise optimizer pass. I think this addresses the issue, but let me know if it can be done better.